bin/schemas/story.json

```diff
@@ -0,0 +1,846 @@
+              "type": "boolean",
+              "title": "Whether it is an external resource"
+            },
+            "creationDate": {
+              "type": "string",
+              "default": "",
+              "title": "Date the media file was uploaded",
+              "format": "date-time"
+            },
+            "posterId": {
+              "type": "integer",
+              "default": 0,
+              "title": "ID of the resource's poster"
+            }
+          }
+        },
+        "overlay": {
+          "type": "object",
+          "$ref": "#/$defs/linearPattern"
+        }
+      }
+    },
+    "shapeElement": {
+      "type": "object",
+      "title": "Shape element",
+      "allOf": [
+        {
+          "$ref": "#/$defs/element"
+        }
+      ]
+    },
+    "pattern": {
+      "type": "object",
+      "oneOf": [
+        { "$ref": "#/$defs/solidPattern" },
+        { "$ref": "#/$defs/linearPattern" }
+      ]
+    },
+    "solidPattern": {
+      "type": "object",
+      "title": "Solid pattern",
+      "required": ["color"],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": ["none", "solid", "linear", "radial"]
+        },
+        "color": {
+          "type": "object",
+          "title": "RGB(A) color",
+          "required": ["r", "g", "b"],
+          "properties": {
+            "r": {
+              "type": "integer",
+              "title": "Red",
+              "minimum": 0,
+              "maximum": 255
+            },
+            "g": {
+              "type": "integer",
+              "title": "Green",
+              "minimum": 0,
+              "maximum": 255
+            },
+            "b": {
+              "type": "integer",
+              "title": "Blue",
+              "minimum": 0,
+              "maximum": 255
+            },
+            "a": {
+              "type": "number",
+              "title": "Alpha",
+              "minimum": 0,
+              "maximum": 1
+            }
+          }
+        }
+      }
+    },
+    "linearPattern": {
+      "type": "object",
+      "title": "Linear color pattern",
+      "required": ["type", "stops"],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": ["none", "solid", "linear", "radial"]
+        },
+        "stops": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "type": "object",
+            "allOf": [
+              {
+                "$ref": "#/$defs/solidPattern"
+              }
+            ],
+            "properties": {
+              "position": {
+                "type": "number",
+                "minimum": 0,
+                "maximum": 1
+              }
+            }
+          }
+        },
+        "alpha": {
+          "type": "number",
+          "minimum": 0,
+          "maximum": 1
+        },
+        "rotation": {
+          "type": "number",
+          "minimum": -360,
+          "maximum": 360
```

Comment:
I think we're restricting it from -359 to 359 instead in the editor.

Reply:
That's good to know. I couldn't find a place where this is done.

Reply:
https://github.com/GoogleForCreators/web-stories-wp/blob/7ffe80de4a77b34d786cd4907b725688fcbc2970/packages/story-editor/src/components/panels/design/sizePosition/usePresubmitHandlers.js#L86